posthog/queries/retention.py

```diff
@@ -0,0 +1,34 @@
+from posthog.models import Filter, Team, Entity, Event
+from typing import Optional, List, Dict, Any
+import datetime
+from datetime import timedelta
+from posthog.queries.base import BaseQuery
+
+
+class Retention(BaseQuery):
+    def calculate_retention(self, filter: Filter, team: Team, start_entity: Optional[Entity] = None, total_days=11):
+        date_from: datetime.datetime = filter.date_from  # type: ignore
+        filter._date_to = (date_from + timedelta(days=total_days)).isoformat()
```

Comment:
Perhaps in a followup refactor since this refactor is overall restructuring. We should standardize how/when to handle filters/params. For example, in this instance above we don't check any conditions and just write in something extra to the filter whereas in trends.py line 298 we do explicit checks before reassigning

Reply:
Yep agree, something I will tackle. It's annoying b/c the way they change the dates isn't consistent